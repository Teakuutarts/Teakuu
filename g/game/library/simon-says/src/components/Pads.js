import React from "react";

import Pad from "./Pad";
import { colors } from "../constants";

export const PadsByComponentName = {
  GreenPad: ({ ...props }) => (
    <Pad m={1} color={colors.green} className="top-left" {...props} />
  ),
  RedPad: ({ ...props }) => (
    <Pad m={1} color={colors.red} className="top-right" {...props} />
  ),
  YellowPad: ({ ...props }) => (
    <Pad m={1} color={colors.yellow} className="bottom-left" {...props} />
  ),
  BluePad: ({ ...props }) => (
    <Pad m={1} color={colors.blue} className="bottom-right" {...props} />
  ),
};

const Pads = ({ pad, onClick }) => {
  const { component } = pad;
  const Composed = PadsByComponentName[component];
  return <Composed {...pad} onClick={onClick} />;
};

export default Pads;
