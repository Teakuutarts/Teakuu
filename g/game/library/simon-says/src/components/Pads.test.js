import React, { Component } from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Pads, { PadsByComponentName } from './Pads';

it('renders Pads correctly', () => {

  const wrapper = shallow(
    <div>
      {Object.keys(PadsByComponentName).map((pad, i) => (
        <Pads pad={pad} key={i} />
      ))}
    </div>
  );

  expect(toJson(wrapper)).toMatchSnapshot();
});

