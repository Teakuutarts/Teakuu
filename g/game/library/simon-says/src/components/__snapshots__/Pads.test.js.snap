exports[`test renders Pads correctly 1`] = `
<div>
  <Pads
    pad="GreenPad" />
  <Pads
    pad="RedPad" />
  <Pads
    pad="YellowPad" />
  <Pads
    pad="BluePad" />
</div>
`;
